File: Chuong 11 - Ke Thua/Ung Dung Quan Ly Thu Vien C++ - C# - Java/Source Code/C#/Quan Ly Doc Gia/QuanLyDocGia/DocGia.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QuanLyDocGia
{
    class DocGia
    {
        protected string HoTen;
	    protected CNgay NgayLapThe = new CNgay();
	    protected int SoThangHieuLuc;

        public void Nhap()
        {
            Console.Write("\nNhap vao ho ten: ");
            HoTen = Console.ReadLine();

            Console.Write("\nNhap vao ngay lam the: ");
            NgayLapThe.Nhap();

            do{
		        Console.Write("\nNhap vao so thang co hieu luc: ");
                SoThangHieuLuc = int.Parse(Console.ReadLine());

		        if(SoThangHieuLuc <= 0)
		        {
			        Console.Write("\nSo thang co hieu luc phai la so duong khac 0. Xin kiem tra lai !");
		        }
	        }while(SoThangHieuLuc <= 0);
            
        }

        public void Xuat()
        {
            Console.Write("\nHo ten: {0}", HoTen);
            
            Console.Write("\nNgay lap the: ");
            NgayLapThe.Xuat();

            Console.Write("\nSo thang co hieu luc: {0}", SoThangHieuLuc);
        }


    }
}

File: Chuong 11 - Ke Thua/Ung Dung Quan Ly Thu Vien C++ - C# - Java/Source Code/Java/Java_QuanLyDocGia/src/java_quanlydocgia/DocGia.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package java_quanlydocgia;

import java.util.Scanner;

/**
 *
 * @author NguyenVietNamSon
 */
public class DocGia {
        protected String HoTen;
	protected CNgay NgayLapThe = new CNgay();
	protected int SoThangHieuLuc;

        public void Nhap()
        {
            Scanner scan = new Scanner(System.in);
            System.out.print("\nNhap vao ho ten: ");
            HoTen = scan.nextLine();

            System.out.print("\nNhap vao ngay lam the: ");
            NgayLapThe.Nhap();

            do{
		        System.out.print("\nNhap vao so thang co hieu luc: ");
                SoThangHieuLuc = Integer.parseInt(scan.nextLine());

		        if(SoThangHieuLuc <= 0)
		        {
			        System.out.print("\nSo thang co hieu luc phai la so duong khac 0. Xin kiem tra lai !");
		        }
	        }while(SoThangHieuLuc <= 0);
            
        }

        public void Xuat()
        {
            System.out.print("\nHo ten: " + String.valueOf(HoTen));
            
            System.out.print("\nNgay lap the: ");
            NgayLapThe.Xuat();

            System.out.print("\nSo thang co hieu luc: " + String.valueOf(SoThangHieuLuc));
        }
}
